src/lib/services/searchUtilActions/AasSearcher.ts

```diff
@@ -0,0 +1,221 @@
+                submodelsSavedInTheRepository,
+            }),
+            stubbedFetch,
+            log ?? Log.createNull(),
+        );
+    }
+
+    async fullSearch(val: string): Promise<AasSearchResult> {
+        const aasIds = await this.handleAasDiscoverySearch(val);
+        if (aasIds && aasIds.length > 1) {
+            return {
+                redirectUrl: `/viewer/discovery?assetId=${val}`,
+                aas: null,
+                aasData: null,
+            };
+        } else {
+            // Check if an AAS ID is found in the Discovery service, or assign the input parameter for further search.
+            // If there is exactly one AAS ID in the aasIds array, use it; otherwise, use the input parameter 'val'.
+            const aasId = aasIds && aasIds.length === 1 ? aasIds[0] : val;
+            const registrySearchResult = await this.handleAasRegistrySearch(aasId);
+
+            let aas: AssetAdministrationShell;
+            if (registrySearchResult != null) {
+                aas = registrySearchResult.registryAas;
+            } else {
+                const aasIdEncoded = encodeBase64(aasId);
+                try {
+                    aas = await this.multipleDataSource.getAasFromDefaultRepository(aasIdEncoded);
+                } catch (e) {
+                    const potentiallyMultipleAas = await this.multipleDataSource.getAasFromAllRepos(aasIdEncoded);
+                    if (potentiallyMultipleAas.length > 1)
+                        return {
+                            redirectUrl: `/viewer/discovery?assetId=${val}`,
+                            aas: null,
+                            aasData: null,
+                        };
+                    aas = potentiallyMultipleAas[0].aas;
+                }
+            }
+
+            const aasData =
+                registrySearchResult?.registryAasData != null
+                    ? {
+                          submodelDescriptors: registrySearchResult.registryAasData.submodelDescriptors,
+                          aasRegistryRepositoryOrigin: registrySearchResult.registryAasData.aasRegistryRepositoryOrigin,
+                      }
+                    : null;
+
+            // If not found: Error: AAS could not be found
+
+            return {
+                redirectUrl: `/viewer/${encodeBase64(aas.id)}`,
+                aas: aas,
+                aasData: aasData,
+            };
+        }
+    }
+
+    /**
+     * Resolves the given AAS ID using the discovery service.
+     *
+     * This function takes an AAS ID and attempts to resolve it using a discovery service.
+     * If the AAS ID is found, it returns the resolved AAS ID.
+     * If not found, it returns `null`.
+     *
+     * @param {string} searchAssetId - The AAS ID to resolve using the discovery service.
+     * @returns {Promise<string | null>} A promise that resolves to the resolved AAS ID as a string, or `null` if the AAS ID is not found.
```

Comment:
Discovery service = Take an Asset ID and look what AAS exist for it
```suggestion
     * Resolves the given Asset ID using the discovery service.
     *
     * This function takes an Asset ID and attempts to resolve it using a discovery service.
     * If the Asset ID is found, it returns the resolved AAS ID.
     * If not found, it returns `null`.
     *
     * @param {string} searchAssetId - The Asset ID to resolve using the discovery service.
     * @returns {Promise<string | null>} A promise that resolves to the AAS ID as a string, or `null` if the Asset ID is not found.
```